### PR TITLE
zero alloc: new payload supported in [@@@zero_alloc check...] to match flags

### PR DESCRIPTION
**Pull request by gretay-js**
In #1936 we added "opt" payload and corresponding flags. This PR enables them to be set on individual files using an attribute at the top level. 

It is a bit confusing, as the last setting in the file applies to the entire file.

Maybe we should just remove "check" payload? We added it in #1294 as a workaround for stdlib, but it has since been removed.  However, I found this top-level attribute useful for quick testing/debugging and tracking down stray allocations, because it overwrites all other settings (command line, env vars, build configurations, etc) and is much cheaper because it doesn't require a rebuilding anything else to enable the check (we commute function summaries unconditionally, this attribute enables the checking).

### CODE REVIEW

**Review by ccasin — APPROVED** [review]
Looks fine, with a minor tweak.

I don't have strong feelings about leaving in `check` or not.

**Comment by ccasin** [comment]
(Also, a test should be added for these, I suppose)

File: ocaml/parsing/builtin_attributes.ml [hunk 1]
```diff
@@ -582,6 +582,9 @@ let zero_alloc_attribute (attr : Parsetree.attribute)  =
   parse_attribute_with_ident_payload attr
     ~name:"zero_alloc" ~f:(function
       | "check" -> Clflags.zero_alloc_check := Clflags.Annotations.Check_default
+      | "check-opt" -> Clflags.zero_alloc_check := Clflags.Annotations.Check_opt_only
+      | "check-all" -> Clflags.zero_alloc_check := Clflags.Annotations.Check_all
+      | "check-none" -> Clflags.zero_alloc_check := Clflags.Annotations.No_check
```
[the remark this answers is not in the document]
Comment (gretay-js) [reply ocaml/parsing/builtin_attributes.ml]: done.

**Comment by ccasin** [comment]
Thanks - looks good to merge, when you're ready.

**Pull request merged**